authors/apps/articles/views.py

```diff
@@ -801,73 +801,60 @@ class SearchForArticles(generics.ListAPIView):
-        payload = {
-            "article_author_hits": 0,
-            "article_body_hits": 0,
-            "article_tag_hits": 0,
-            "article_description_hits": 0,
-            "article_title_hits": 0,
-            "total_articles": 0,
-            "articles": []
-        }
-        search_fields = []
-        if (my_query_params):
-            search_fields = my_query_params.split(",")
+        search_string = request.query_params.get('query', None)
+        payload = []
         if not search_string:
-            payload = {
+            reply = {
                 "message": "Please enter a search string."
             }
-            return Response(payload, status=status.HTTP_400_BAD_REQUEST)
-
-        if('title' in search_fields) or (not search_fields):
-            query = Article.objects.filter(title__icontains=search_string)
-            payload["article_title_hits"] = len(query)
-            query_fields = list(set(query_fields + list(query)))
-        if('body' in search_fields) or (not search_fields):
-            query = Article.objects.filter(body__icontains=search_string)
-            payload["article_body_hits"] = len(query)
-            query_fields = list(set(query_fields + list(query)))
-        if('author' in search_fields) or (not search_fields):
-            query = Article.objects.filter(
-                author__user__username__icontains=search_string
+            return Response(reply, status=status.HTTP_400_BAD_REQUEST)
+
+        if slug == "author":
+            payload = Article.objects.filter(
+                author__user__username__icontains=search_string, published=True,
+                activated=True
             )
-            payload["article_author_hits"] = len(query)
-            query_fields = list(set(query_fields + list(query)))
-        if('description' in search_fields) or (not search_fields):
-            query = Article.objects.filter(
-                description__icontains=search_string
+        if slug == "title":
+            payload = Article.objects.filter(
+                title__icontains=search_string, published=True,
+                activated=True
+            )
+        if slug == "body":
+            payload = Article.objects.filter(
+                body__icontains=search_string, published=True,
+                activated=True
+            )
+        if slug == "description":
+            payload = Article.objects.filter(
+                description__icontains=search_string, published=True,
+                activated=True
             )
-            payload["article_description_hits"] = len(query)
-            query_fields = list(set(query_fields + list(query)))
-
         matching_tags = []
-        if('tags' in search_fields) or (not search_fields):
-            matching_tags = list(Tag.objects.filter(
-                tag__icontains=search_string
-            ))
-        if matching_tags:
-            count = 0
+        matching_tags = list(Tag.objects.filter(
```

Comment:
@Etomovich, you do not need to convert this to a list, you can work with the queryset as is.

Reply:
> @Etomovich, you do not need to convert this to a list, you can work with the queryset as is.

Noted